src/FilterButtonList.js

```diff
@@ -0,0 +1,152 @@
+import F from 'futil-js'
+import _ from 'lodash/fp'
+import React from 'react'
+import { observer } from 'mobx-react'
+import {
+  Dynamic,
+  Flex,
+  CheckButton as DefaultCheckButton,
+  Modal as DefaultModal,
+  Popover as DefaultPopover,
+} from './layout'
+import DefaultIcon from './DefaultIcon'
+import DefaultMissingTypeComponent from './DefaultMissingTypeComponent'
+import InjectTreeNode from './utils/injectTreeNode'
+import styles from './styles'
+
+let FilterButtonItem = ({
+  node,
+  tree,
+  fields,
+  mapNodeToProps,
+  mapNodeToDescription,
+  mapNodeToLabel,
+  Button,
+  CheckButton,
+  MissingTypeComponent,
+  Modal,
+}) => {
+  let mappedProps = mapNodeToProps(node, fields)
+  let modal = F.stateLens(React.useState(false))
+  let title = // we really need a title, so here's every possible fallback
+    _.get('label', mappedProps) ||
+    mapNodeToLabel(node, fields) ||
```

Comment:
i don't think we need to explicitly support mapNodeToLabel and mapNodeToDescription, just mapNodeToProps is probably fine

Reply:
also mergeOverAll can help compose mapNodeToProps functions